src/query_ctrl.js

```diff
@@ -9,13 +9,22 @@ export class GenericDatasourceQueryCtrl extends QueryCtrl {
     this.scope = $scope;
     this.target.target = this.target.target || 'select metric';
     this.target.type = this.target.type || 'timeseries';
-    this.target.data = this.target.data || null;
+    this.target.data = this.target.data || "";
+
+    // special handling when in table panel
+    if (!this.target.format) {
+      this.target.format = this.panelCtrl.panel.type === 'table' ? 'table' : 'timeseries';
+    }
+
+    this.formats = [{ text: 'Time series', value: 'timeseries' }, { text: 'Table', value: 'table' }];
+    this.showJSON = false;
   }
 
   getOptions(query) {
     return this.datasource.metricFindQuery(query || '');
   }
 
+  // not used
```

Comment:
if not used, can we toss it away?

Reply:
Didn't want to do that since its still in upstream? If remove then we can also scrap the part in the query editor.

Reply:
Ok, let's skip this for now.